sys/arm64/arm64/machdep.c

```diff
@@ -567,6 +568,15 @@ try_load_dtb(caddr_t kmdp)
 	vm_pointer_t dtbp;
 
 	dtbp = MD_FETCH(kmdp, MODINFOMD_DTBP, vm_offset_t);
+#ifdef __CHERI_PURE_CAPABILITY__
+	if (dtbp != (vm_pointer_t)NULL) {
```

Comment:
I don't think you need the cast on NULL?

Reply:
You do, otherwise you get `<source>:9:15: warning: comparison between pointer and integer ('vm_pointer_t' (aka 'unsigned __intcap') and 'void * __attribute__((cheri_no_provenance))') [-Wpointer-integer-compare]`

(and equivalently for vanilla it will warn)